drivers/adodb-mysqli.inc.php

```diff
@@ -231,9 +231,13 @@ function _connect($argHostname = null,
 
 		// SSL Connections for MySQLI
 		if ($this->ssl_key || $this->ssl_cert || $this->ssl_ca || $this->ssl_capath || $this->ssl_cipher) {
+
 			mysqli_ssl_set($this->_connectionID, $this->ssl_key, $this->ssl_cert, $this->ssl_ca, $this->ssl_capath, $this->ssl_cipher);
-			$this->socket = MYSQLI_CLIENT_SSL;
-			$this->clientFlags = MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT;
+
+			// Default client flag to ssl with verify if not set.
+			if (0 === $this->clientFlags) {
```

Comment:
I don't think this check is correct. 

The client flags can be set to a number of different values independently (binary flag), so it can be != 0 without MYSQLI_CLIENT_SSL being set.

I believe you need to check for the appropriate value(s) with a binary and, e.g. `$this->clientFlags & MYSQLI_CLIENT_SSL` instead.

That being said, reading the [documentation](https://www.php.net/manual/en/mysqli.real-connect.php#mysqli.real-connect.flags), I must admit I'm a bit confused with these SSL flags, and which one(s) should be used for the above-mentioned check... 

 * `MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT` this one is clear
 *  `MYSQLI_CLIENT_SSL_VERIFY_SERVER_CERT` is only referenced in the predefined constants page, but not on mysqli_real_connect page.
 * `MYSQLI_CLIENT_SSL` means *Use SSL (encryption)* but [here](https://www.php.net/manual/en/mysqli.constants.php#constant.mysqli-client-ssl) it is stated that it *is set internally in the MySQL client library* (without specifying how or when) and that *should not be set by application programs*

Reply:
Updated to check for set bitwise options MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT and MYSQLI_CLIENT_SSL_VERIFY_SERVER_CERT  but I am still unsure if I should add MYSQLI_CLIENT_SSL as an additional check...